src/main/java/games/strategy/triplea/delegate/MustFightBattle.java

```diff
@@ -2691,11 +2692,18 @@ private void endBattle(final IDelegateBridge bridge) {
     clearWaitingToDie(bridge);
     m_isOver = true;
     m_battleTracker.removeBattle(this);
+
+    // Must clear transportedby for allied air on carriers for both attacking units and retreating units
     final CompositeChange clearAlliedAir =
         clearTransportedByForAlliedAirOnCarrier(m_attackingUnits, m_battleSite, m_attacker, m_data);
     if (!clearAlliedAir.isEmpty()) {
       bridge.addChange(clearAlliedAir);
     }
+    final CompositeChange clearAlliedAirRetreated =
+        clearTransportedByForAlliedAirOnCarrier(m_attackingUnitsRetreated, m_battleSite, m_attacker, m_data);
+    if (!clearAlliedAirRetreated.isEmpty()) {
+      bridge.addChange(clearAlliedAirRetreated);
+    }
```

Comment:
This is the main change which is adding an additional check for retreating units.